### ISSUE TITLE
Sidecar's Eureka Instance Config Has Wrong Port

### ISSUE THREAD

**Issue by tkvangorder** [spring-cloud/spring-cloud-netflix#896]
We are using Brixton.M5 and I noticed that the port being registered with Eureka is incorrect. The SidecarConfiguration registers its own copy of the EurekaInstanceConfigBean. 

The server port is injected as a value into the configuration class but never used. Instead it only consults the "sidecar" properties. 

This should be a pretty easy fix: If the sidecar properties are not defined, fallback to logic that is similar to that found in EurekaClientAutoConfiguration.eurekaInstanceConfigBean.

@spencergibb Let me know if you would like help with this one, I can put a pull request together.

**Comment by spencergibb** [comment]
@tkvangorder Don't you want the sidecar properties port to be the one registered with eureka? If `server.port` is used it will register the port of the sidecar, which doesn't make sense to me.

**Comment by tkvangorder** [comment]
Yeah, I definitely think it makes sense to use the sidecar properties. It would be nice if the sidecar is registering with eureka: 

It would fail fast if the sidecar properties are not set.

Or

The fallback is to use the sidecar's port when the sidecar properties are not set.
 
Without setting the sidecar properties for the node app listening on port 3000 (and the sidecar listening on port 8082), our metadata ends up looking like this:


```
<instance>
  .
  .
  <port enabled="true">0</port>
  <securePort enabled="false">443</securePort>
  .
  .
  <homePageUrl>/info</homePageUrl>
  <statusPageUrl>http://TVANGO1001034.Build.Internal:8080/info</statusPageUrl>
  <healthCheckUrl>http://TVANGO1001034.Build.Internal:8080/health</healthCheckUrl>

</instance>
```

It's easy enough to fix but it was behavior that changed between Brixton.M3 and Brixton.M5

**Comment by tkvangorder** [comment]
OK, I think I tracked down the change to the health and status pages: 

in M3, the server port was used to construct the URI: 

```
		@Value("${server.port:${SERVER_PORT:${PORT:8080}}}")
		private int serverPort = 8080;

.
.
.
			String scheme = config.getSecurePortEnabled() ? "https" : "http";
			config.setStatusPageUrl(scheme + "://" + config.getHostname() + ":"
					+ this.serverPort + config.getStatusPageUrlPath());
			config.setHealthCheckUrl(scheme + "://" + config.getHostname() + ":"
					+ this.serverPort + config.getHealthCheckUrlPath());

```

in M5, the management port was used to construct the URI and the server port is not used.  

```
		@Value("${server.port:${SERVER_PORT:${PORT:8080}}}")
		private int serverPort = 8080;

		@Value("${management.port:${MANAGEMENT_PORT:${PORT:8080}}}")
		private int managementPort = 8080;

.
.
			config.setStatusPageUrl(scheme + "://" + config.getHostname() + ":"
					+ this.managementPort + config.getStatusPageUrlPath());
			config.setHealthCheckUrl(scheme + "://" + config.getHostname() + ":"
					+ this.managementPort + config.getHealthCheckUrlPath());

```

In  EurekaClientAutoConfig, the management port falls back to the server port, before defaulting to 8080

```
	@Value("${management.port:${MANAGEMENT_PORT:${server.port:${SERVER_PORT:${PORT:8080}}}}}")
```

Regardless, trying to figure out which sidecar properties override the normal eureka.instance properties is rough. 

It might be better to just drop the configuration of the EurekanstanceConfigBean from the SidecarConfiguration and instead document how to use the "StatusPageUrlPath", "HealthCheckUrlPath", management port, and server port to correctly define these things or if a sidecar needs to get creative, they can just define their own instance.

**Comment by spencergibb** [comment]
I've updated managementPort to behave like the one in `EurekaClientAutoConfig`. A `@Primary` `EurekanstanceConfigBean` should work. More docs are needed for sure.

**Comment by yankhonskikendaxa** [comment]
> @tkvangorder Don't you want the sidecar properties port to be the one registered with eureka? If `server.port` is used it will register the port of the sidecar, which doesn't make sense to me.

I had the same issue. I have spring gateway and spring registry. I have a python-service and java sidecar for it. I want registry to see only sidecar, and to know nothing about the python app.

I want nobody to know about python app, but the only sidecar (which has zuul proxy) will forward request to the sidecar.

**Comment by spencergibb** [comment]
Sidecar doesn't work that way. It does not proxy requests to the service. It only registers the port of the external process (in your case python).
